packages/caliper-samples/src/contract/scalardl/src/main/java/com/example/contract/simple/SimpleTransfer.java

```diff
@@ -0,0 +1,41 @@
+package com.example.contract.simple;
+
+import com.scalar.ledger.asset.Asset;
+import com.scalar.ledger.contract.Contract;
+import com.scalar.ledger.ledger.Ledger;
+import java.util.Optional;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+public class SimpleTransfer extends Contract {
+
+  @Override
+  public JsonObject invoke(Ledger ledger, JsonObject argument, Optional<JsonObject> properties) {
+    String srcAccountId = argument.getString("account1");
+    String dstAccountId = argument.getString("account2");
+    // open.js sends amount as number but transfer.js sends it as string...
+    int amount = Integer.parseInt(argument.getString("money"));
+
+    Optional<Asset> srcAsset = ledger.get(srcAccountId);
+    Optional<Asset> dstAsset = ledger.get(dstAccountId);
+
+    JsonObject srcData = srcAsset.get().data();
+    JsonObject dstData = dstAsset.get().data();
+    int srcBalance = srcData.getInt("balance");
+    int dstBalance = dstData.getInt("balance");
+    srcBalance -= amount;
+    dstBalance += amount;
+
+    JsonObjectBuilder newSrcData = Json.createObjectBuilder();
```

Comment:
Instead of recreating from scratch, you can do `Json.createObjectBuilder(srcData)` and update with the new balance.
Please update others as well.

Reply:
Thanks, I fixed it. I didn't know the interface well...